api/src/test/java/uk/gov/dhsc/htbhf/claimant/message/processor/DetermineEntitlementMessageProcessorTest.java

```diff
@@ -146,16 +224,162 @@ void shouldUpdateClaimWhenClaimantIsNotEligible(EligibilityStatus eligibilitySta
+        verify(childDateOfBirthCalculator).hadChildrenUnderFourAtGivenDate(TWO_CHILDREN, currentPaymentCycleStartDate);
+        verifyZeroInteractions(childDateOfBirthCalculator);
+    }
+
+    @ParameterizedTest(name = "Qualifying benefit status={0}")
+    @ValueSource(strings = {"CONFIRMED", "NOT_CONFIRMED"})
+    void shouldExpireClaimWhenClaimantWasPregnantButNoLongerWithChildrenInPreviousCycle(
+            QualifyingBenefitEligibilityStatus qualifyingBenefitEligibilityStatus) {
+        //Given
+        List<LocalDate> previousCycleChildrenDobs = SINGLE_THREE_YEAR_OLD;
+        List<LocalDate> currentCycleChildrenDobs = NO_CHILDREN;
+        LocalDate previousCycleExpectedDeliveryDate = EXPECTED_DELIVERY_DATE_IN_TWO_MONTHS;
+        LocalDate currentCycleExpectedDeliveryDate = NOT_PREGNANT;
+        DetermineEntitlementMessageContext context = buildMessageContext(
+                previousCycleExpectedDeliveryDate,
+                currentCycleExpectedDeliveryDate,
+                previousCycleChildrenDobs,
+                currentCycleChildrenDobs);
+        given(messageContextLoader.loadDetermineEntitlementContext(any())).willReturn(context);
+        given(pregnancyEntitlementCalculator.isEntitledToVoucher(any(), any())).willReturn(false);
+        //Had children in last cycle who are under 4 at the start of the current payment cycle.
+        given(childDateOfBirthCalculator.hadChildrenUnder4AtStartOfPaymentCycle(any())).willReturn(true);
+        //Children in previous cycle are now be over 4.
+        given(childDateOfBirthCalculator.hadChildrenUnderFourAtGivenDate(any(), any())).willReturn(false);
+
+        //Eligibility - any status that isn't ELIGIBLE has the same effect here.
+        EligibilityAndEntitlementDecision decision = aDecisionWithStatusAndChildren(INELIGIBLE, qualifyingBenefitEligibilityStatus, currentCycleChildrenDobs);
+        given(eligibilityAndEntitlementService.evaluateExistingClaimant(any(), any(), any())).willReturn(decision);
+
+        //Current payment cycle voucher entitlement mocking
+        Message message = aValidMessageWithType(DETERMINE_ENTITLEMENT);
+
+        //When
+        MessageStatus messageStatus = processor.processMessage(message);
+
+        //Then
+        runCommonVerifications(context, decision, message, messageStatus, ClaimStatus.EXPIRED);
+        verify(childDateOfBirthCalculator).hadChildrenUnder4AtStartOfPaymentCycle(context.getPreviousPaymentCycle());
+        verify(pregnancyEntitlementCalculator).isEntitledToVoucher(NOT_PREGNANT, context.getCurrentPaymentCycle().getCycleStartDate());
+        verifyZeroInteractions(determineEntitlementNotificationHandler);
+    }
+
+    @ParameterizedTest(name = "Qualifying benefit status={0}")
+    @ValueSource(strings = {"CONFIRMED", "NOT_CONFIRMED"})
+    void shouldExpireClaimWhenClaimantWasPregnantButNoLongerWithNoChildrenInPreviousCycle(
```

Comment:
I'm also unsure whether this and the previous test are correct. If the claimant has no children, was previously pregnant and is now no longer pregnant, should we expire them with no email? I'm double checking with Karen...

Reply:
I have double checked this with Karen and this is correct for the scenario where the claimant was pregnant but is now no longer pregnant (but would have been in the new cycle). There is another story (HTBHF-1503) to do the scenario where they haven't told DWP about their children yet and they are 8 (or 12 weeks can't remember which) after their due date.